src/chain/store/chain_store.rs

```diff
@@ -594,6 +590,82 @@ where
 
         Ok(digest)
     }
+
+    /// Get the [`TipsetKeys`] for a given epoch. The [`TipsetKeys`] may **not** be null.
+    pub fn get_epoch_tsk(
+        &self,
+        tipset: Arc<Tipset>,
+        round: ChainEpoch,
+    ) -> Result<TipsetKeys, Error> {
+        let ts = self
+            .tipset_by_height(round, tipset, true)
+            .map_err(|e| Error::Other(format!("Could not get tipset by height {e:?}")))?;
+        Ok(ts.key().clone())
+    }
+
+    /// Gets look-back tipset for block validations.
+    ///
+    /// The look-back tipset for a round is the tipset with epoch `round -
+    /// chain_finality`. Chain finality is usually 900. The given is a
+    /// reference point in the blockchain such that the look-back tipset can
+    /// be found by tracing the `parent` pointers.
+    pub fn get_lookback_tipset_for_round(
```

Comment:
I don't have a firm understanding of what a lookback tipset is, or why you'd need one - is there a way of clarifying?

Reply:
It's the tipset and state-tree root for a given epoch.

Reply:
Err, for `epoch - 900`, usually.

Reply:
I'll add that it also includes the state-tree root:
> Gets look-back tipset (and state-root of that tipset) for block validations.
> 
> The look-back tipset for a round is the tipset with epoch `round - chain_finality`

Can it be improved further?